protocol/protocol.go

```diff
@@ -611,18 +634,54 @@ func VerifyReply(replyBytes, publicKey []byte, nonce [NonceSize]byte) (time uint
 	for len(path) > 0 {
 		pathElementIsRight := index&1 == 0
 		if pathElementIsRight {
-			hashNode(&hash, hash[:], path[:sha512.Size])
+			hashNode(&hash, hash[:32], path[:32])
 		} else {
-			hashNode(&hash, path[:sha512.Size], hash[:])
+			hashNode(&hash, path[:31], hash[:32])
 		}
 
 		index >>= 1
-		path = path[sha512.Size:]
+		path = path[31:]
 	}
 
-	if !bytes.Equal(hash[:], root) {
-		return 0, 0, errors.New("protocol: calculated tree root doesn't match signed root")
+	if !bytes.Equal(hash[:32], root) {
+		return mjd.Mjd{}, 0, errors.New("protocol: calculated tree root doesn't match signed root")
 	}
 
 	return midpoint, radius, nil
 }
+
+// EncapsulatePacket applies the encapsulation.
```

Comment:
Perhaps this could be clarified to:
```go
// EncapsulatePacket creates a UDP/TCP payload for a Roughtime message.
```
based on https://tools.ietf.org/html/draft-ietf-ntp-roughtime-01#section-6
I was confused at first by the "12", expecting 4+4=8, but that appears to be a spec typo. Fix: https://github.com/aanchal4/draft-roughtime/pull/26